akka-http/src/main/scala/akka/http/scaladsl/server/directives/RouteDirectives.scala

```diff
@@ -49,6 +49,22 @@ trait RouteDirectives {
   def complete(m: => ToResponseMarshallable): StandardRoute =
     StandardRoute(_.complete(m))
 
+  /**
+   * Completes the request using the given arguments.
+   *
+   * @group route
+   */
+  def complete[T](status: StatusCode, v: => T)(implicit m: ToEntityMarshaller[T]): StandardRoute =
+    StandardRoute(_.complete((status, v)))
+
+  /**
+   * Completes the request using the given arguments.
+   *
+   * @group route
+   */
+  def complete[T](status: StatusCode, headers: immutable.Seq[HttpHeader], v: T)(implicit m: ToEntityMarshaller[T]): StandardRoute =
```

Comment:
```suggestion
  def complete[T](status: StatusCode, headers: immutable.Seq[HttpHeader], v: => T)(implicit m: ToEntityMarshaller[T]): StandardRoute =
```

Reply:
Incidentally, while I think it's good to have those as by-name for consistency, it's less likely that they are used in the wrong way (compared to the one-argument `complete`). The one argument `complete` could be used like that:


```scala
complete {
  // run some side-effects in here
}
```

but that's less likely with multiple arguments.